### ISSUE TITLE
hitresult and kiai images not preloading in taiko

### ISSUE THREAD

**Issue by pavlukivan** [ppy/osu-stable-issues#360]
I play osu!taiko, and when I get the first judgement of the certain type for the session the game often hangs, I assume it happens because the image isn't preloaded. For example, it happens when I hit the first finisher (big note) of the game, or when I get the first 100 or 300.

**Comment by pavlukivan** [comment]
Other users (at least two, one of them on two different PCs) report the same.

**Comment by peppy** [comment]
Please provide your skin which causes such issues. Alternatively, please state that you are testing using the default skin.

**Comment by pavlukivan** [comment]
I currently use the following overrides for taiko hitcircles:

[skin.zip](https://github.com/ppy/osu-stable-issues/files/4203983/skin.zip)

**Comment by peppy** [comment]
Can you reproduce the same issue with the default skin?

**Comment by peppy** [comment]
Hit results are correctly preloaded. Maybe this is sound related? Could you please confirm whether it is an issue for the default skin, and provide your skin's samples too?

**Comment by pavlukivan** [comment]
The issue happens with 100 specifically, not just any result, so I'm not sure how it can be related to sounds.

**Comment by peppy** [comment]
Please answer my questions. Can you reproduce with the default skin?

**Comment by pavlukivan** [comment]
it rarely happens so i can't consistently reproduce it at all

**Comment by peppy** [comment]
Closing this one as as far as I can tell they are fine here.

**Comment by pavlukivan** [comment]
other people report it as well, i couldnt record it but it definitely happens occasionally @peppy 
ill send discord screenshots in a bit

**Comment by pavlukivan** [comment]
![image](https://user-images.githubusercontent.com/9590981/77801453-756cbb00-70ab-11ea-84b9-434ac05a4ad4.png)
![image](https://user-images.githubusercontent.com/9590981/77801503-92a18980-70ab-11ea-83a8-099708ef57ef.png)
^ in Russian - "the most irritating shit is when you launch osu! and play the first map, it freezes for about 0.5s after the first hit"

**Comment by pavlukivan** [comment]
https://www.twitch.tv/videos/571893236 here on 03:08 I miss on an easy pattern because that was the first map I played the game and it lagged

**Comment by pavlukivan** [comment]
Honestly, even if they should be preloading, they actually don't. Drawing them outside the screen on skin load or whatever bandaid solution like that is fine because the fact is the current solution doesn't work.

**Comment by pavlukivan** [comment]
![image](https://user-images.githubusercontent.com/9590981/77803036-cb8f2d80-70ae-11ea-880f-95ce725cf888.png)
"i always have that when getting the 1st 100 in session, probably because of the 18 frame animation loading for the first time"

![image](https://user-images.githubusercontent.com/9590981/77803059-dd70d080-70ae-11ea-8129-9998a33c40bd.png)
"ssd. i mean it happens with ssd as well"

**Comment by pavlukivan** [comment]
just got a report thats even worse, that doesnt happen to me personally
![image](https://user-images.githubusercontent.com/9590981/77803930-ee224600-70b0-11ea-8841-0e79ec6c9f4a.png)

**Comment by pavlukivan** [comment]
![image](https://user-images.githubusercontent.com/9590981/77804232-a18b3a80-70b1-11ea-9491-58e4a5904c45.png)

i will stop posting others' reports at this point cuz i think thats enough. I'll post a video if someone records it.

**Comment by pavlukivan** [comment]
https://drive.google.com/file/d/1FfVRFZ2mNmLTgDHSyXL2adrgCiQmCl42/view a recording. It's barely noticeable, but it definitely lagged a little.

**Comment by peppy** [comment]
If people are suddenly getting changed behaviour, it is 100% driver related. So help by tracking down which hardware and driver is responsible so we can either report to the device manufacturer or attempt to patch against it.

**Comment by pavlukivan** [comment]
The recording above uses the default skin just as you requested. I'll resend it - https://drive.google.com/file/d/1FfVRFZ2mNmLTgDHSyXL2adrgCiQmCl42/view

**Comment by peppy** [comment]
Interested to hear reports from other users on this issue. Need to know GPU hardware and driver versions, as well.

**Comment by pavlukivan** [comment]
In my case it's Intel HD Graphics 4600 with drivers from 2018/08/16 (20.19.15.5058), I'll try asking other players

It's definitely an old issue, as I've found this forum post describing the issue: https://osu.ppy.sh/community/forums/topics/487300 (except it only describes it for taiko-specific skin, even though it happens always)

**Comment by pavlukivan** [comment]
Alright, made a skin so that you can reproduce it even with a better machine. It makes it very clear the images aren't being preloaded until they need to be drawn for the first time. https://drive.google.com/file/d/1rtkTYHbNF0F5N8aaNwx05pYmq7CI8Syv/view?usp=sharing

**Comment by peppy** [comment]
Thanks for your persistence. I think I have found a potential cause for this, could you please test cutting edge?

**Comment by pavlukivan** [comment]
Nope, it still lags a lot on first hit with the skin I provided, and not on the consecutive ones

**Comment by pavlukivan** [comment]
Nvm, it fixed itself after a restart somehow, I'll reopen if it reappears

**Comment by pavlukivan** [comment]
Sadly the issue still persists, I can confirm it now - https://www.youtube.com/watch?v=JcKk0hE7VVA?t=21

Not only it lagged on the first 300 (happened too early so it wasn't captured), it also lagged on the first finisher 300 I got. I have an RTX 2070 and a Ryzen 3900 now so it certainly isn't a hardware program, I didn't notice any lags later on in the game.

**Comment by peppy** [comment]
Are you basing this all off one single play? Did you not play from the last fix until now?

We can't act on such on-a-whim performance reports so please provide more data. ie. a video that shows this happening every time you start the game in a single video.

**Comment by pavlukivan** [comment]
I did not play from the last fix until now because of lags I mentioned previously, it was simply impossible to play without random misses caused by frame+input drops. Other users have also reported this not being fixed - but it's incredibly hard to gather data from the community. I will be sure to provide you with more clips when I get them, the bug doesn't happen consistently but it does happen relatively often.

**Comment by peppy** [comment]
It's also the first large hit in that beatmap. Are you sure that's not related?

**Comment by pavlukivan** [comment]
It isn't related to streaming at all. And the fact it's the first large hit is precisely why it's lagged, as I mentioned before it also lagged on the first note on the map.

**Comment by peppy** [comment]
The fact that you are able to reproduce this on two systems while I am unable to across my available hardware and VMs says there may be something else at play here. Can you turn off any overlays, other applications, etc. and try again?

**Comment by pavlukivan** [comment]
I am not using any overlays (specifically disabled discord/etc), and the bug doesn't happen consistently, neither it happened consistently on the old hardware.

**Comment by pavlukivan** [comment]
Just caught it again - https://www.youtube.com/watch?v=moP8SCtzCyo&t=55

Gameplay starts at 0:55, it lags on every hitresult load - 300, 100, finisher

**Comment by pavlukivan** [comment]
I can try hooking some low-level IO/graphics calls and see what happens when such a lag occurs, would that trigger the anticheat?
